lib/temporal/workflow/state_manager.rb

```diff
@@ -322,10 +322,9 @@ def parse_failure(failure, default_exception_class = StandardError)
         when :application_failure_info
           exception_class = safe_constantize(failure.application_failure_info.type)
           exception_class ||= default_exception_class
-          details = from_payloads(failure.application_failure_info.details)
+          message = from_details_payloads(failure.application_failure_info.details)
```

Comment:
nit: the details might contain more than just a message, it can be a serialized error attributes depending on the error class

Reply:
@antstorm I thought/hoped the same at first, but alas, we only serialize the message.
https://github.com/coinbase/temporal-ruby/blob/7d3f1bf82b434af54d2f39956a4263433c9719ed/lib/temporal/client/serializer/failure.rb#L16

Reply:
Yeah, something we'll need to fix